modules/hal_silabs/simplicity_sdk/CMakeLists.txt

```diff
@@ -23,8 +23,15 @@ set(BLOBS_DIR      ${ZEPHYR_HAL_SILABS_MODULE_DIR}/zephyr/blobs/simplicity_sdk)
 # respectively.
 string(TOUPPER ${CONFIG_SOC_SERIES} SILABS_DEVICE_FAMILY)
 
-# Get SoC series number, i.e. translate e.g. efr32bg22 -> 22
-string(SUBSTRING ${CONFIG_SOC_SERIES} 7 2 SILABS_DEVICE_FAMILY_NUMBER)
+if("${CONFIG_SOC_PART_NUMBER}" MATCHES "MGM24")
+  set(SILABS_DEVICE_IS_MODULE TRUE)
```

Comment:
Please introduce a new Kconfig symbol for this, and set it for the MGM240 series, rather than hard-coding it in CMake.